### DISCUSSION TITLE
Highlight specific line numbers in py-editor.

### DISCUSSION THREAD

**Discussion by ntoll** [pyscript/pyscript#2066] [Proposals]
Requested by @marieroald as part of the TurtleThread.com efforts. They would like to use PyScript (particularly MicroPython) to make their documentation interactive. One of the aspects of their documentation is, as you build up an example within a tutorial, the changes in the code are highlighted so the learner can see what the difference is as they learn new capabilities.

This is an awesome thing to do..!

Can we add a feature such that the `py-editor` / `mpy-editor` understands a attribute like `highlight="2, 4, 5"`:

```
<script type="mpy-editor" highlight="2, 3, 4">
import foo
print("Hello from highlighted line 2")
# This is highlighted because it is line 3
# So is this (line 4)
print("I'm not highlighted")
</script>
```

For an example of how this looks please see: https://turtlethread.com/en/intro/introduction.html

If the user starts to type in the resulting editor, the highlighting for the specified lines should disappear.

**Comment by WebReflection** [comment]
this looks like either a feature request for Codemirror or the introduction of a Codemirror plugin ... has anyone looked at that?

if nothing already out there, we need to hook into codemirror logic and events (within lines) and I hope that won't mess up everything else as I don't know all codemirror internals (but I am not against this feature request).

**Reply by ntoll** [reply]
I've just had a look into this. 

There are several discussions online for how to highlight a range of lines, and the solutions all feel bespoke / hacky or only work with older versions of CodeMirror. As a result, there's not a simple/obvious way to move forward. I assumed (incorrectly) that such a feature was already built into CodeMirror's API - but this not the case (although I notice concepts such as a [selection](https://codemirror.net/docs/ref/#state.EditorSelection) could perhaps be re-purposed for this).
